python/ray/data/tests/test_backpressure_policies.py

```diff
@@ -365,11 +365,16 @@ def test_e2e_backpressure(self):
         )
         # We can buffer at most 2 blocks.
         # Thus the producer should be backpressured after producing 2 blocks.
-        # Note, although block 2 is backpressured, but producer_timestamps[2] has
-        # been generated before the backpressure. Thus we assert
-        # producer_timestamps[2] < consumer_timestamps[0] < producer_timestamps[3].
+        # Note, although block 2 is backpressured, producer_timestamps[2] has
+        # been generated before the backpressure.
+        # Thus producer_timestamps[2] < consumer_timestamps[0].
+        # For block 3, it should be generated after the first consumer task
+        # is scheduled. But there is a delay between the consumer task is scheduled
+        # and the consumer task starts to run. Thus the order of
+        # producer_timestamps[3] and consumer_timestamps[0] is not deterministic.
+        # To avoid flakiness, we check consumer_timestamps[0] < producer_timestamps[4].
```

Comment:
@raulchen to confirm my understanding, we are guaranteed that `consumer_timestamps[0] < producer_timestamps[4]` since block 4 will only be generated after the second consumer task is scheduled, which only occurs after the first consumer task consuming block 3 concludes?

Reply:
yes